src/adaptations/device-layer/LwIP/WarmSupport.cpp

```diff
@@ -358,27 +383,66 @@ PlatformResult AddRemoveThreadAddress(InterfaceType inInterfaceType, const Inet:
 
 #if WARM_CONFIG_SUPPORT_THREAD_ROUTING
 
-#error "Weave Thread router support not implemented"
-
 PlatformResult StartStopThreadAdvertisement(InterfaceType inInterfaceType, const Inet::IPPrefix &inPrefix, bool inStart)
 {
-    // TODO: implement me
+    if (inInterfaceType != kInterfaceTypeThread) { return kPlatformResultFailure; }
+    if ((inPrefix.Length & 7) != 0) { return kPlatformResultFailure; }
+
+    // Thread Stack should attempt to become a router if params are set, so nothing special needed
+
+    return kPlatformResultSuccess;
 }
 
 #endif // WARM_CONFIG_SUPPORT_THREAD_ROUTING
 
 #if WARM_CONFIG_SUPPORT_BORDER_ROUTING
 
-#error "Weave border router support not implemented"
-
 PlatformResult AddRemoveThreadRoute(InterfaceType inInterfaceType, const Inet::IPPrefix &inPrefix, RoutePriority inPriority, bool inAdd)
 {
-    // TODO: implement me
+    WEAVE_ERROR err = WEAVE_NO_ERROR;
+    otError otErr;
+    otExternalRouteConfig routeConfig;
+
+    int ot_priority = OT_ROUTE_PREFERENCE_HIGH;
+    switch (inPriority)
+    {
+        case kRoutePriorityLow:
```

Comment:
switch should align with  case, see this example
https://github.com/openweave/openweave-core/blob/master/src/lib/profiles/data-management/Current/SubscriptionClient.cpp#L171